modules/task/internal/stats/metrics.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package stats
+
+import "github.com/uber-go/tally"
+
+const (
+	// TagModule is module tag for metrics
+	TagModule = "module"
+	// TagType is either request or response
+	TagType = "type"
+)
+
+// HTTPTags creates metrics scope with defined tags
+var (
+	// TaskTags creates metrics scope with defined tags
+	TaskTags = map[string]string{
+		TagModule: "task",
+	}
+
+	// TaskExecutionCount counts number of executions
+	TaskExecutionCount tally.Counter
+	// TaskPublishCount counts number of tasks enqueued
+	TaskPublishCount tally.Counter
+	// TaskExecuteFail counts number of tasks failed to execute
+	TaskExecuteFail tally.Counter
+	// TaskPublishFail counts number of tasks failed to enqueue
+	TaskPublishFail tally.Counter
+	// TaskExecutionTime is a turnaround time for execution
+	TaskExecutionTime tally.Timer
+	// TaskPublishTime is a publish time for tasks
+	TaskPublishTime tally.Timer
```

Comment:
Missing the always-crucial time spent in queue, which will require you to define an envelope for your tasks.

If you haven't read something similar already, consider reading [this](https://www.vividcortex.com/resources/queueing-theory) for a quick-and-dirty introduction to queueing theory.